uitests/tests/org.jboss.tools.fuse.ui.bot.tests/src/org/jboss/tools/fuse/ui/bot/tests/RouteContextSupportTest.java

```diff
@@ -0,0 +1,125 @@
+import java.util.Collection;
+
+import org.eclipse.reddeer.common.wait.TimePeriod;
+import org.eclipse.reddeer.common.wait.WaitUntil;
+import org.eclipse.reddeer.eclipse.ui.console.ConsoleView;
+import org.eclipse.reddeer.eclipse.ui.views.contentoutline.ContentOutline;
+import org.eclipse.reddeer.junit.runner.RedDeerSuite;
+import org.eclipse.reddeer.requirements.cleanworkspace.CleanWorkspaceRequirement;
+import org.jboss.tools.fuse.reddeer.ProjectTemplate;
+import org.jboss.tools.fuse.reddeer.editor.CamelEditor;
+import org.jboss.tools.fuse.reddeer.projectexplorer.CamelProject;
+import org.jboss.tools.fuse.reddeer.utils.LogChecker;
+import org.jboss.tools.fuse.reddeer.utils.ProjectFactory;
+import org.jboss.tools.fuse.reddeer.wizard.CamelXmlFileWizard;
+import org.jboss.tools.fuse.ui.bot.tests.utils.EditorManipulator;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.eclipse.reddeer.swt.api.TreeItem;
+import org.eclipse.reddeer.swt.condition.ShellIsAvailable;
+
+/**
+ * @author fpospisi
+ */
+@RunWith(RedDeerSuite.class)
+public class RouteContextSupportTest {
+	public static final String PROJECT_NAME = "routeContextSupportTest";
+	public static final String ROUTES_NAME = "camel-routes.xml";
+	
+	/**
+	 * <p>
+	 * routeContext support test
+	 * </p>
+	 * <b>Steps</b>
+	 * <ol>
+	 * <li>Create new project: STANDALONE - KARAF - CBR_SPRING.</li>
+	 * <li>In source tab change XML to external <routeContext> "camel-routes.xml" reference.</li>
+	 * <li>Create new XML - camel-routes.xml.</li>
+	 * <li>Edit 'Source' in camel-routes.xml.</li>
+	 * <li>Check camel-routes.xml 'Design' tab for containing items setted up in source. </li>
+	 * <li>Run as Local Camel Context (without tests) and check for Fuse related errors.</li>
+	 * </ol>
+	 */
+	
+	@Before
+	/*
+	 * Create new project.
+	 * */
+	public void setWorkspace() {
+		ProjectFactory.newProject(PROJECT_NAME).deploymentType(STANDALONE).runtimeType(KARAF).template(ProjectTemplate.CBR_SPRING).create();
+	}
+	
+	@After
+	public void cleanWorkspace() {
+		new CleanWorkspaceRequirement().fulfill();
+	}
+	
+	@Test
+	public void testRouteContextSupport() {		
+		/*
+		 * Change content in 'Source'tab for main file.
+		 */
+		CamelEditor.switchTab("Source");
+		EditorManipulator.copyFileContentToCamelXMLEditor("resources/routeContextSupportCamelContext.xml");
+		
+		/*
+		 * Create new XML file 'route-context.xml'.
+		 */
+		create_new_file(ROUTES_NAME);
+
+		/*
+		 * Change content in 'Source' tab for 'route-context.xml'.
+		 */
+		CamelEditor.switchTab("Source");
+		EditorManipulator.copyFileContentToCamelXMLEditor("resources/routeContextSupportRouteContext.xml");				
+		
+		/*
+		 * Check if design tab contains <routeContext>
+		 */
+		check_outline(); 
+		
+		/*
+		 * Run as Local Camel Context (without tests).
+		 */
+		CamelProject project = new CamelProject(PROJECT_NAME);
+		project.openCamelContext("camel-context.xml");
```

Comment:
camel-context.xml could be also static final class variable as you already have e.g. `ROUTES_NAME`